wrapper.go

```diff
@@ -0,0 +1,119 @@
+package sfxlambda
+
+import (
+	"context"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-lambda-go/lambdacontext"
+	"github.com/signalfx/golib/datapoint"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"strings"
+	"time"
+)
+
+type HandlerWrapper struct {
+	lambda.Handler
+	defaultDimensions  map[string]string
+	notColdStart       bool
+}
+
+func (hw *HandlerWrapper) Invoke(ctx context.Context, payload []byte) ([]byte, error) {
+	hw.defaultDimensions = defaultDimensions(ctx)
+	dps := []*datapoint.Datapoint {hw.invocationsDatapoint(),}
+	start := time.Now()
+	responseBytes, err := hw.Handler.Invoke(ctx, payload)
+	dps = append(dps, hw.durationDatapoint(time.Since(start)))
+	if err != nil {
+		dps = append(dps, hw.errorsDatapoint())
+	}
+	if !hw.notColdStart {
+		dps = append(dps, hw.coldStartsDatapoint())
+		hw.notColdStart = true
+	}
+	hw.SendDatapoints(ctx, dps)
+	return responseBytes, err
+}
+
+type dimensions map[string]string
+
+func Start(handler interface{}) {
+	lambda.StartHandler(&HandlerWrapper{Handler: lambda.NewHandler(handler)})
+}
+
+// SendDatapoints sends custom metrics to SignalFx.
+func (hw *HandlerWrapper) SendDatapoints(ctx context.Context, dps []*datapoint.Datapoint) {
+	sendDatapoints(ctx, dps)
```

Comment:
Why don't you put the addition of the `hw.defaultDimensions` here in this method and then have your`*Datapoint()` methods below just create the datapoint and remove all of the duplicated default dimension handling?